drf_extra_fields/fields.py

```diff
@@ -30,9 +30,18 @@
 
 
 class Base64FieldMixin(object):
-    ALLOWED_TYPES = NotImplemented
-    INVALID_FILE_MESSAGE = NotImplemented
-    INVALID_TYPE_MESSAGE = NotImplemented
+    @property
+    def ALLOWED_TYPES(self):
+        raise NotImplementedError
+
+    @property
+    def INVALID_FILE_MESSAGE(self):
+        raise NotImplementedError
+
+    @property
+    def INVALID_TYPE_MESSAGE(self):
+        raise NotImplementedError
+
```

Comment:
This pr has two purpose. 
1. Fixing the NotImplementedError.
2. Extending the usage of type and message attributes of the Base64FieldMixin. This is out of the pr's main topic. Having 2 different pr's could have been better.

All the method names are lower-case in the project. I can see that you named them all upper-case to be able to support old versions. But this is against the followed naming convention.

Separate method like `get_allowed_types` which returns the ALLOWED_TYPES by default can be used. That would allow people to override allowed types inside a method if necessary. 

Do you have such a need or though somebody might need this?

Reply:
I did not get how did you figure out the second purpose. Converting `NotImplemented` to `NotImplementedError` is all I did. 

The correct way of marking an attribute as not implemented is making it a property method which raises `NotImplementedError`. Yes, it conflicts with naming convention, but I cannot see a way that we can provide backwards compatibility and avoid this violation.

Reply:
You are right. I misunderstood your intention. Looks like this is the only way to be able to provide backwards compatibility. 

Thank you.